ktor-network/common/test/io/ktor/network/sockets/tests/TCPSocketTest.kt

```diff
@@ -306,4 +306,32 @@ class TCPSocketTest {
             socket.awaitClosed()
         }
     }
+
+    @Test
+    fun testAutoFlush() = testSockets { selector ->
+        val tcp = aSocket(selector).tcp()
+        val server: ServerSocket = tcp.bind("127.0.0.1", port = 0)
+
+        val serverConnectionPromise = async {
+            server.accept()
+        }
+
+        val clientConnection = tcp.connect("127.0.0.1", port = server.port)
+        val serverConnection = serverConnectionPromise.await()
+        val serverInput = serverConnection.openReadChannel()
+
+        val writeChannel = clientConnection.openWriteChannel(autoFlush = true)
+        writeChannel.writeStringUtf8("Hello, world\n")
+        delay(100) //wait for read to be available
+        assertTrue(serverInput.availableForRead > 0)
+        val message = serverInput.readUTF8Line()
+        assertEquals("Hello, world", message)
```

Comment:
We can avoid delays here by reading the line and letting it suspend.

```suggestion
        val message = serverInput.readUTF8Line()
        assertEquals("Hello, world", message)
```

Reply:
I've changed it to your suggestion, but this would lead to tests running into a longer timeout should they fail in the future